packages/app-desktop/app.ts

```diff
@@ -201,6 +205,15 @@ class Application extends BaseApplication {
 		}
 	}
 
+	public updateMenuBar() {
+		// macOS disallows the removal of the menu bar
+		if (shim.isMac()) return;
+
+		const window = bridge().window();
+		window.setAutoHideMenuBar(Setting.value('hideMenuBar'));
+		window.setMenuBarVisibility(!Setting.value('hideMenuBar'));
+	}
```

Comment:
We have a `MenuBar` component to handle the menu. Would you mind moving the logic over there? If you need some help implementing this please let me know.

Essentially you'll need to add this to `mapStateToProps`:

`showMenuBar: state.settings.showMenuBar,`

Then check the value of `props.showMenuBar` from the component and hide/show it based on this.

The reason for doing it that way is to encapsulate all menu bar related logic in the same place, and to avoid the need to sync the settings and the state in the middleware.

Reply:
Thanks for the review! Makes sense - I'll try implementing it this way and reply back here.

Reply:
Moved the implementation to the `MenuBar` component in the latest commit. That simplifies things a lot!